pkgs/blast_repo/lib/src/tweaks/auto_publish_tweak.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../exact_file_tweak.dart';
+
+final _instance = AutoPublishTweak._();
+
+class AutoPublishTweak extends ExactFileTweak {
+  factory AutoPublishTweak() => _instance;
+
+  AutoPublishTweak._()
+      : super(
+          id: 'auto-publish',
+          description:
+              'configure a github action to enable package auto-publishing',
+          filePath: '.github/workflows/publish.yaml',
+        );
+
+  @override
+  bool get stable => false;
+
+  @override
+  String expectedContent(String repoSlug) {
+    final org = repoSlug.split('/').first;
+
+    // Substitute the org value for the pattern '{org}'.
+    return publishContents.replaceAll('{org}', org);
+  }
+
+  @override
+  List<String> performAdditionalFixes(Directory checkout, String repoSlug) {
+    var results = <String>[];
+
+    // Update the readme to include contribution + publishing info.
+    const tag = 'Contributions, PRs, and publishing';
+
+    var readmeFile = File(path.join(checkout.path, 'README.md'));
+
+    if (readmeFile.existsSync()) {
+      var contents = readmeFile.readAsStringSync();
+
+      if (!contents.contains(tag)) {
+        var newContents = '${contents.trimRight()}\n\n$readmeSection';
+        readmeFile.writeAsStringSync(newContents);
+        results.add('README.md updated with contribution and publishing info.');
+      }
+    }
+
+    return results;
+  }
+}
+
+const publishContents = r'''
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    tags: [ 'v[0-9]+.[0-9]+.[0-9]+*' ]
+
+jobs:
+  publish:
+    if: github.repository_owner == '{org}'
+    uses: devoncarew/firehose/.github/workflows/publish.yaml@main
```

Comment:
Shouldn't we have a thing under dart-lang ecosystem here?

Reply:
Yes, for sure. I'd like to iterate on the workflow a bit (my understanding of how this would work has changed a bunch in the last month or two). Then plan to move the tooling - w/ some cleanup - into dart-lang/ecosystem.

Reply:
I don't think this introduces risk here - I'm the only one who's contributed to that repo. wdyt?

Reply:
okay...